docs/content/development/cache/create.md

```diff
@@ -0,0 +1,247 @@
+
+---
+
+## Step 1: Creating the Module
+
+The first step is to create and prepare your module for development. You can refer to the [Create a Module](../modules/create.mdx) documentation, specifically the first three steps, to learn how to create your module.
+
+Once you're done, you can start implementing your module.
+
+---
+
+## Step 2: Create the Service
+
+Create the file `src/services/memcached-cache.ts` which will hold your cache service. Note that the name of the file is recommended to be in the format `<service_name>-cache`. So, if you’re not integrating `memcached`, you should replace the name with what’s relevant for your module.
+
+Add the following content to the file:
+
+```ts title=src/services/memcached-cache.ts
+import { ICacheService } from "@medusajs/types"
+
+class MemcachedCacheService implements ICacheService {
+  async get<T>(key: string): Promise<T> {
+    throw new Error("Method not implemented.")
+  }
+  async set(
+    key: string, 
+    data: unknown, 
+    ttl?: number
+  ): Promise<void> {
+    throw new Error("Method not implemented.")
+  }
+  async invalidate(key: string): Promise<void> {
+    throw new Error("Method not implemented.")
+  }
+}
+
+export default MemcachedCacheService
+```
+
+This creates the class `MemcachedCacheService` that implements the `ICacheService` interface imported from `@medusajs/types`. Feel free to rename the class to what’s relevant for your cache service.
+
+In the class, you must implement three methods: `get`, `set`, and `invalidate`. You’ll learn what each of these methods do in the upcoming section.
+
+---
+
+## Step 3: Implement the Methods
+
+### constructor
+
+The `constructor` method of a service allows you to prepare any third-party client or service necessary to be used in other methods. It also allows you to get access to the module’s options which are typically defined in `medusa-config.js`, and to other services and resources in the Medusa backend using [dependency injection](../fundamentals/dependency-injection.md).
+
+Here’s an example of how you can use the `constructor` to create a memcached instance and save the module’s options:
+
+```ts title=src/services/memcached-cache.ts
+import { ICacheService } from "@medusajs/types"
+import Memcached from "Memcached"
+
+const DEFAULT_CACHE_TIME = 30
+
+export type MemcachedCacheModuleOptions = {
+  /**
+   * Time to keep data in the cache (in seconds)
+   */
+  ttl?: number
+
+  /**
+    * Allow passing the configuration for Memcached client
+    */
+  location: Memcached.Location
+  options?: Memcached.options
+}
+
+class MemcachedCacheService implements ICacheService {
+  protected readonly memcached: Memcached
+  protected readonly TTL: number
+
+  constructor(
+    {
+      // inject services through dependency injection
+      // for example the Redis Cache module injects
+      // redis connection
+      cacheRedisConnection,
```

Comment:
suggestion: I think that it is a good idea to explain here how a resource from the DI container can be accessed, but I would maybe use another example (since these 2 are kinda competing technologies).

Also, I created the connection in the constructor for sake of simplicity but since we are already explaining how to use DI, we can have a proper Memcached loader and pull the connection from the DI container as we do with the Redis module:
```
constructor(
    { memcachedConnection },
    options: MemcachedCacheModuleOptions
){
  this.memcached =  memcachedConnection
  this.TTL = options.ttl || DEFAULT_CACHE_TIME
}
```
But then we would need to expand a bit more about loaders. I'm fine with both approaches.

Reply:
Good point regarding competing technologies, but I wouldn't expand into loaders as it's outside the scope of this guide in particular. For now I just replaced it with `logged` as an example of dependency injection.

Reply:
Sounds good 👍